src/openrct2-ui/UiContext.macOS.mm

```diff
@@ -137,10 +138,12 @@ void OpenURL(const std::string& url) override
                 panel.directoryURL = [NSURL fileURLWithPath:directory];
                 if ([panel runModal] == NSModalResponseCancel)
                 {
+                    SDL_RaiseWindow(window);
```

Comment:
This doesn't depend on condition above, please move this line higher to above condition check

Reply:
The code needs to run after `runModal` has been called on the panel or the window won't activate after the panel has been dismissed

Should I set the string to a variable, and return that?
i.e.
```c++
std::string returnString = std::string();
if ([panel runModal] == NSModalResponseCancel)
{
    returnString = std::string();
}
else
{
    returnString = panel.URL.path.UTF8String;
}
SDL_RaiseWindow(window);
return returnString;
```